File: internal/app/commands/help/installation.go
package help

import (
	"fmt"
	"github.com/bwmarrin/discordgo"
	"meteor-bot/internal/app/common"
)

type InstallationCommand struct {
	common.BaseCommand
}

func NewInstallationCommand() *InstallationCommand {
	return &InstallationCommand{
		BaseCommand: *common.NewCommandBuilder().
			SetName("installation").
			SetDescription("Tells someone to read the installation guide").
			Build(),
	}
}

func (c *InstallationCommand) Build() *discordgo.ApplicationCommand {
	return &discordgo.ApplicationCommand{
		Name:        c.Name(),
		Description: c.Description(),
		Options: []*discordgo.ApplicationCommandOption{
			{
				Name:        "member",
				Description: "The member to tell to read the installation guide",
				Type:        discordgo.ApplicationCommandOptionUser,
				Required:    true,
			},
		},
	}
}

func (c *InstallationCommand) Handle(s *discordgo.Session, i *discordgo.InteractionCreate) {
	targetMember := i.ApplicationCommandData().Options[0].UserValue(s)
	c.HandleInteractionRespond(s, i, &discordgo.InteractionResponse{
		Type: discordgo.InteractionResponseChannelMessageWithSource,
		Data: &discordgo.InteractionResponseData{
			Embeds: []*discordgo.MessageEmbed{
				{
					Title:       "Read the Installation Guide",
					Description: fmt.Sprintf("%s The installation guide answers your question, please read it.", targetMember.Mention()),
					Color:       common.EmbedColor,
				},
			},
			Components: []discordgo.MessageComponent{
				discordgo.ActionsRow{
					Components: []discordgo.MessageComponent{
						discordgo.Button{
							Label: "Guide",
							Style: discordgo.LinkButton,
							URL:   "https://meteorclient.com/faq/installation",
						},
					},
				},
			},
		},
	})
}

File: src/commands/help/installation.rs
use crate::{Ctx, Error, config::constants::EMBED_COLOR};
use poise::{CreateReply, serenity_prelude as serenity};
use serenity::{
    builder::{CreateActionRow, CreateButton, CreateEmbed},
    model::user::User,
    prelude::Mentionable,
};

/// Tells someone to read the installation guide
#[poise::command(slash_command, category = "Help")]
pub async fn installation(
    ctx: Ctx<'_>,
    #[description = "User to direct to the installation guide"] user: User,
) -> Result<(), Error> {
    let embed = CreateEmbed::default()
        .title("Read the Installation Guide")
        .description(format!(
            "{} The installation guide answers your question, please read it.",
            user.mention()
        ))
        .color(EMBED_COLOR);

    let button = CreateButton::new_link("https://meteorclient.com/faq/installation").label("Guide");
    let action_row = CreateActionRow::Buttons(vec![button]);

    ctx.send(CreateReply::default().embed(embed).components(vec![action_row]))
        .await?;

    Ok(())
}
